coordinator-service/src/disk-coordinator.ts

```diff
@@ -117,8 +187,12 @@ export class DiskCoordinator implements Coordinator {
         //
         // Return false if contributor trying to lock unverified chunk or
         // if verifier trying to lock verified chunk.
-        //
+        // Also return false if contributor has already contributed
+        // to the chunk.
         const verifier = this.db.verifierIds.includes(participantId)
+        if (!verifier && hasContributed(participantId, chunk)) {
```

Comment:
Good catch, we shouldn't allow locking for someone who has successfully contributed in the past!